code/modules/mob/living/simple_animal/hostile/slime.dm

```diff
@@ -59,95 +58,117 @@
 	if(bodytemperature < 273.15)
 		calm()
 
+
+/mob/living/simple_animal/hostile/slime/MoveToTarget()
+	..()
+	if(target && target.Adjacent(src))
+		forceMove(get_turf(target))
+
+/mob/living/simple_animal/hostile/slime/AttackingTarget()
+	forceMove(get_turf(target))
+	..()
+
 /mob/living/simple_animal/hostile/slime/proc/calm()
 	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/mob/living/simple_animal/hostile/slime/proc/calm() called tick#: [world.time]")
+	var/calmed_type = /mob/living/carbon/slime
 	switch(colour)
```

Comment:
It's funny, if the colours perfectly matched the typepath you could just use

text2path("/mob/living/carbon/slime/" + colour)